src/schoology-extractor/schoology_extractor/mapping/submissions.py

```diff
@@ -58,8 +58,11 @@ def map_to_udm(submissions_df: pd.DataFrame) -> pd.DataFrame:
 
     df["SourceSystem"] = constants.SOURCE_SYSTEM
     df["EntityStatus"] = constants.ACTIVE
-    df["SubmissionStatus"] = df["late"].apply(lambda x: 'late' if x == 1 else '')
-    df["SubmissionStatus"] = df["draft"].apply(lambda x: 'draft' if x == 1 else x)
+    df["SubmissionStatus"] = 'on-time'
+    for index, row in df.iterrows():
+        row["SubmissionStatus"] = 'late' if row["late"] == 1 else row["SubmissionStatus"]
+        row["SubmissionStatus"] = 'draft' if row["draft"] == 1 else row["SubmissionStatus"]
```

Comment:
Iterating rows should be avoided when possible.

```suggestion
       def _get_status(row: pd.Series):
           if row["late"] == 1:
               return "late"
           if row["draft"] == 1:
               return "draft"
           return "on-time"

        row["SubmissionStatus"] = df.apply(_get_status, axis=1)
```

Apply should be much faster. There are other possible optimizations but I haven't quite understood them yet.

Reply:
✅